### PR TITLE
posix: Implement pthread_equal function

### LINKED ISSUE #59942: posix: implement `pthread_equal()`
There are currently 20 missing POSIX functions with trivial implementations, `pthread_equal()` is one of them.

> pthread_equal - compare thread IDs
> 
> This function shall compare the thread IDs t1 and t2.

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_equal.html

See RFC #51211 for more info.

### PR DESCRIPTION
**Pull request by jaiiarora**
Patch implements pthread_equal function.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/59942

### CODE REVIEW

**Review by cfriedt — COMMENTED** [review]
@jaiiarora - Looks good, but can you please also add a second commit with tests in `tests/posix/common/src/pthread.c`?

Also, can you update your commit message? Technically, `pthread_equal()` was implemented, but your change converts it from a static inline function to a regular function.

**Comment by jaiiarora** [comment]
Okay, I will work on it

**Comment by jaiiarora** [comment]
add required changes @cfriedt

**Comment by jaiiarora** [comment]
Please let me know if anything  is missing @cfriedt. I will work on it

File: tests/posix/common/src/pthread.c [hunk 1]
```diff
@@ -30,6 +30,7 @@
 K_THREAD_STACK_ARRAY_DEFINE(stack_e, N_THR_E, STACKS);
 K_THREAD_STACK_ARRAY_DEFINE(stack_t, N_THR_T, STACKS);
 K_THREAD_STACK_ARRAY_DEFINE(stack_1, 1, 32);
+K_THREAD_STACK_DEFINE(stack_p, STACKS);
```
Comment (cfriedt) [inline tests/posix/common/src/pthread.c:33]: This can be removed.

File: tests/posix/common/src/pthread.c [hunk 2]
```diff
@@ -771,3 +772,58 @@ ZTEST(posix_apis, test_posix_pthread_barrier)
...
+	printk("parent thread_id is %u\n", pthread_self());
+
+	/* both should have different id */
+	if (pthread_equal(thread_id, pthread_self()))
+		zassert_true(false, "pthread_equal failed");
+
+	ret = pthread_attr_destroy(&attr);
+	zassert_equal(ret, 0, "pthread_attr_destroy failed");
+
+	printk("pthread_equal test ok\n");
```
Comment (cfriedt) [inline tests/posix/common/src/pthread.c:828]:
```suggestion
ZTEST(posix_apis, test_pthread_equal)
{
	zassert_true(pthread_equal(pthread_self(), pthread_self()));
	zassert_false(pthread_equal(pthread_self(), (pthread_t)4242);
```

**Review by cfriedt — CHANGES_REQUESTED** [review]

File: tests/posix/common/src/pthread.c (hunk 1 above)
Comment (jaiiarora) [reply tests/posix/common/src/pthread.c]: done

File: tests/posix/common/src/pthread.c (hunk 2 above)
Comment (jaiiarora) [reply tests/posix/common/src/pthread.c]: added requested changes

File: lib/posix/pthread.c [hunk 3]
```diff
@@ -848,4 +848,9 @@ static int posix_thread_pool_init(void)
 	return 0;
 }
 
+int pthread_equal(pthread_t pt1, pthread_t pt2)
+{
+	return (pt1 == pt2);
```
Comment (cfriedt) [inline lib/posix/pthread.c:853]: nit: no brackets needed here, but I'm not going to stop this from going in now. If there is another review that requires another spin, please fix this too.

**Review by cfriedt — APPROVED** [review]
LGTM - thanks @jaiiarora 👍

**Review by ycsin — APPROVED** [review]

**Pull request merged**